### PR TITLE
Import open-telemetry/opentelemetry-proto submodule and generate protobuf bindings locally

### LINKED ISSUE #793: Consider generating Go types for OTLP in this repo
This will allow us to delete Go code from the `opentelemetry-proto` and make that repo cleaner in general.

See https://github.com/open-telemetry/opentelemetry-proto/issues/79#issuecomment-637881895 and the rest of the thread there.

### PR DESCRIPTION
**Pull request by evantorrie**
Fixes #793 and will allow opentelemetry-proto repository maintainers to remove the generated Go code from their repo.

   1. OTLP .proto sources added as git submodule at `internal/opentelemetry-proto`. 
   1. Generated protobuf files committed to `internal/opentelemetry-proto-gen`.
   1. References to Go module `github.com/open-telemetry/opentelemetry-proto` now changed to reference internal package `go.opentelemetry.io/otel/internal/opentelemetry-proto-gen`.

In normal workflow, the git submodule should *not* be populated. Only do so with `git submodule update --init` when a new version of the protobuf sources are released in the [opentelemetry-proto repository](https://github.com/open-telemetry/opentelemetry-proto) and you need to force the workflow to regenerate updated Go bindings in `internal/opentelemetry-proto-gen`.

### CODE REVIEW

File: example/otel-collector/go.sum [hunk 1]
```diff
@@ -711,8 +708,6 @@ github.com/onsi/gomega v1.9.0/go.mod h1:Ho0h+IUsWyvy1OpqCwxlQ/21gkhVunqlU8fDGcoT
 github.com/open-telemetry/opentelemetry-collector v0.3.0 h1:/i106g9t6xNQ4hOAuczEwW10UX+S8ZnkdWDGawo9fyg=
 github.com/open-telemetry/opentelemetry-collector v0.3.0/go.mod h1:c5EgyLBK6FoGCaJpOEQ0j+sHqmfuoRzOm29tdVA/wDg=
 github.com/open-telemetry/opentelemetry-proto v0.3.0/go.mod h1:PMR5GI0F7BSpio+rBGFxNm6SLzg3FypDTcFuQZnO+F8=
```
[the remark this answers is not in the document]
Comment (evantorrie) [reply example/otel-collector/go.sum:710]: The example here is [pinned to v0.3.0](https://github.com/open-telemetry/opentelemetry-go/blob/dd79483e20e449c586ef573249afbd76cb71fe93/example/otel-collector/go.mod#L5) in the `go.mod` file, although the k8s example references [`opentelemetry-collector:0.5.0`](https://github.com/open-telemetry/opentelemetry-go/blob/dd79483e20e449c586ef573249afbd76cb71fe93/example/otel-collector/k8s/otel-collector.yaml#L117)

Looks like the module is pulled in only to get the semantic attribute name for *service.name*, which should now be available locally to this repo in the `standards` module.

I'll create a separate PR to eliminate this dependency.

[the remark this answers is not in the document]
Comment (lizthegrey) [reply example/otel-collector/go.sum:710]: Thank you!

[the remark this answers is not in the document]
Comment (evantorrie) [reply example/otel-collector/go.sum:710]: See https://github.com/open-telemetry/opentelemetry-go/pull/943

**Comment by MrAlias** [comment]
@lizthegrey for backstory: https://github.com/open-telemetry/opentelemetry-proto/issues/79#issuecomment-637881895

and from that the collector solution: https://github.com/open-telemetry/opentelemetry-collector/pull/1037
it is just a manually run command

**Comment by lizthegrey** [comment]
> @lizthegrey for backstory: [open-telemetry/opentelemetry-proto#79 (comment)](https://github.com/open-telemetry/opentelemetry-proto/issues/79#issuecomment-637881895)
> 
> and from that the collector solution: [open-telemetry/opentelemetry-collector#1037](https://github.com/open-telemetry/opentelemetry-collector/pull/1037)
> it is just a manually run command

Ah, excellent, so a similar solution of checking the generated .pb.go files in subject to normal code review, but not run automatically from circleci and instead run by hand. We should consider upstreaming the Github Action so that they get the benefit of it too!

**Comment by yurishkuro** [comment]
🎉 🎉 🎉

**Comment by yurishkuro** [comment]
Recommendation: in Jaeger we have a separate CI step "check-proto" that re-runs the .pb.go file generation and verifies that checked-in version is the same as would-be-generated one. It catches issues when something in the toolchain changes, like the version of generator plugin.

**Comment by jmacd** [comment]
Thanks @evantorrie

**Pull request merged**